src/test/java/org/javarosa/core/model/SelectChoiceTest.java

```diff
@@ -62,4 +69,54 @@ public void value_should_continue_being_an_empty_string_after_deserialization()
         deserializedScenario.next();
         assertThat(deserializedScenario.getQuestionAtIndex().getChoice(0).getValue(), is(""));
     }
+
+    @Test
+    public void getChild_returnsNamedChild_whenChoicesAreFromSecondaryInstance() {
+        setUpSimpleReferenceManager(r("external-select-geojson.xml").getParent(), "file");
+
+        Scenario scenario = Scenario.init("external-select-geojson.xml");
+        assertThat(scenario.choicesOf("/data/q").get(1).getChild("geometry"), CoreMatchers.is("104 0.5 0 0"));
+        assertThat(scenario.choicesOf("/data/q").get(1).getChild("special-property"), CoreMatchers.is("special value"));
+    }
+
+    @Test
+    public void getChild_returnsNull_whenChoicesAreFromSecondaryInstance_andRequestedChildDoesNotExist() {
+        setUpSimpleReferenceManager(r("external-select-geojson.xml").getParent(), "file");
+
+        Scenario scenario = Scenario.init("external-select-geojson.xml");
+        assertThat(scenario.choicesOf("/data/q").get(1).getChild("non-existent"), CoreMatchers.is(nullValue()));
+    }
+
+    @Test
+    public void getChild_updates_whenChoicesAreFromRepeat() throws IOException {
```

Comment:
Dude. You could, like, have a repeat that has a field called `geometry`. And then build a select with a `map` appearance on top of that. I don't know why exactly but that makes my brain explode a little bit. 🤯

I almost only populated the `item` field in the case of an external secondary instance because this kind of very dynamic situation scared me. But I can't actually think of a case where it wouldn't work.